### PR TITLE
XRT-1135 branch2

### PR DESCRIPTION
**Pull request by SteveOss**
Update to cache file and socket descriptors on private logger struct.
Fixes to work on AzureSphere.

### CODE REVIEW

**Review by BinduRao2018 — COMMENTED** [review]
Have added 2 minor comments

File: include/iot/data.h [hunk 1]
```diff
@@ -1121,6 +1121,19 @@ extern char * iot_data_to_json (const iot_data_t * data);
...
+/**
+ * @brief  Convert data to json string, writing to provided buffer
+ *
+ * An initial string buffer and size is provided. If the buffer is too small a new buffer will
+ * be allocated, in which case the returned pointer will not be the same as the provided buffer.
+ *
+ * @param  data  Input data
+ * @param  buff  Output buffer
+ * @param  size  Output buffer size
+ * @return       Output buffer or a new buffer if provided buffer to small
```
[the remark this answers is not in the document]
Comment (SteveOss) [reply include/iot/data.h]: Have fixed

File: include/iot/logger.h [hunk 2]
```diff
@@ -22,68 +22,73 @@ extern "C" {
...
+  IOT_LOG_DEBUG,      /**< Debug, Information, Warning and Error logging */
+  IOT_LOG_TRACE       /**< Trace, Debug, Information, Warning and Error logging */
 } iot_loglevel_t;
 
-/** Alias for IOT logger function pointer */
-typedef void (*iot_log_function_t) (struct iot_logger_t * logger, iot_loglevel_t level, uint64_t timestamp, const char * message);
+/**
+ * Log type enumeration
+ */
+typedef enum iot_logger_type_t
```
[the remark this answers is not in the document]
Comment (SteveOss) [reply include/iot/logger.h:44]: Have added logger functions to header as are actually required if not using as a component.

**Review by BinduRao2018 — APPROVED** [review]
LGTM

**Comment by BinduRao2018** [comment]
@SteveOss , Can you please merge as I don't have the permission?

**Pull request merged**